### PR TITLE
[jit] Added a new operator for concat that takes in variadic parameters

### PR DESCRIPTION
**Pull request by navahgar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #59881 [jit] Added a pass to transform aten::cat ops to prim::Concat op with variable number of inputs
* **#59880 [jit] Added a new operator for concat that takes in variadic parameters**

Differential Revision: [D29277877](https://our.internmc.facebook.com/intern/diff/D29277877)

### CODE REVIEW

File: torch/csrc/jit/runtime/register_prim_ops.cpp [hunk 1]
```diff
@@ -745,6 +745,19 @@ RegisterOperators reg(
            handler(ss.str());
          },
          aliasAnalysisSpecialCase()),
+     OperatorGenerator(
+         TORCH_SELECTIVE_SCHEMA("prim::Concat(...) -> Tensor"),
```
Comment (eellison) [inline torch/csrc/jit/runtime/register_prim_ops.cpp:749]: So, the convention here is that it's (Tensors..., dim) right ?

File: torch/csrc/jit/runtime/register_prim_ops.cpp [hunk 2]
```diff
@@ -745,6 +745,19 @@ RegisterOperators reg(
...
+         [](Stack* stack) {
+           auto num_inputs = pop(stack).toInt();
+           auto dim = pop(stack).toInt();
+           std::vector<at::Tensor> inputs(num_inputs - 1);
+           for (int i = 0; i < num_inputs - 1; ++i) {
+             inputs[i] = pop(stack).toTensor();
+           }
+           std::reverse(inputs.begin(), inputs.end());
```
Comment (eellison) [inline torch/csrc/jit/runtime/register_prim_ops.cpp:757]: would there be a way of writing this so we don't need to have the reverse ?

File: torch/csrc/jit/runtime/register_prim_ops.cpp (hunk 1 above)
Comment (eellison) [inline torch/csrc/jit/runtime/register_prim_ops.cpp:749]: I think we probably want `prim::ConstantConcat`, following along the lines of `prim::ConstantChunk`. I'm not sure though.

**Review by eellison — COMMENTED** [review]
LGTM - this PR doesn't turn it on but i'd be curious to hear from @Krovatkin @ZolotukhinM @bertmaher on their opinions here. 

Also cc @haolu, should we get this turned on in static runtime? might reduce overhead of constructing input list

**Review by eellison — APPROVED** [review]

File: torch/csrc/jit/runtime/register_prim_ops.cpp (hunk 1 above)
Comment (navahgar) [reply torch/csrc/jit/runtime/register_prim_ops.cpp:753]: Yes. Added a comment above.

Comment (navahgar) [reply torch/csrc/jit/runtime/register_prim_ops.cpp:753]: I'm open to changing that, if everyone prefers that.

**Comment by navahgar** [comment]
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29277877).

**Comment by navahgar** [comment]
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29277877).

**Comment by navahgar** [comment]
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29277877).

**Comment by navahgar** [comment]
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29277877).

**Pull request closed without merging**